tests/e2e/run-compatibility.sh

```diff
@@ -0,0 +1,174 @@
+# shellcheck source=../../tests/e2e/lib.sh
+source "$ROOT/tests/e2e/lib.sh"
+
+run_compatibility_tests() {
+    if [[ "$#" -ne 2 ]]; then
+        info "run_compatibility_tests() Args: $*, Numargs: $#"
+        die "missing args. usage: run_compatibility_tests <central_version> <sensor_version>"
+    fi
+
+    local central_version="$1"
+    local sensor_version="$2"
+
+    short_central_tag="$(shorten_tag "${central_version}")"
+    short_sensor_tag="$(shorten_tag "${sensor_version}")"
+
+    info "Starting go compatibility tests with central v${short_central_tag} and sensor v${short_sensor_tag}"
+
+    junit_wrap CentralSensorVersionCompatibility "central: ${short_central_tag}, sensor: ${short_sensor_tag}" "" \
+        _run_compatibility_tests "${central_version}" "${sensor_version}" "${short_central_tag}" "${short_sensor_tag}"
+}
+
+_run_compatibility_tests() {
+    local central_version="$1"
+    local sensor_version="$2"
+    local short_central_tag="$3"
+    local short_sensor_tag="$4"
+    local compatibility_dir="compatibility-test-central-v${short_central_tag}-sensor-v${short_sensor_tag}"
+
+    info "Starting test (go compatibility test Central v${short_central_tag}, Sensor v${short_sensor_tag})"
+
+    require_environment "KUBECONFIG"
+
+    export_test_environment
+    ci_export CENTRAL_PERSISTENCE_NONE "true"
+
+    export SENSOR_HELM_DEPLOY=true
+    export ROX_ACTIVE_VULN_REFRESH_INTERVAL=1m
+    export ROX_NETPOL_FIELDS=true
+    export ROX_SENSOR_UPGRADER_ENABLED="false"
+
+    test_preamble
+    setup_deployment_env false false
+    remove_existing_stackrox_resources
+    setup_default_TLS_certs
+    info "Creating mocked compliance operator data for compliance v1 tests"
+    "$ROOT/tests/complianceoperator/create.sh"
+
+    deploy_stackrox_with_custom_central_and_sensor_versions "${central_version}" "${sensor_version}"
+    echo "Stackrox deployed with Central version - ${central_version}, Sensor version - ${sensor_version}"
+
+    rm -f FAIL
+
+    prepare_for_endpoints_test
+
+    # Give some time for stackrox to be reachable
+    wait_for_api
+
+    info "API version compatibility tests"
+    if pr_has_label "ci-release-build"; then
+        echo "Running version compatibility tests in release mode"
+        export GOTAGS=release
+    fi
+    kubectl -n stackrox get pods
+    make -C tests compatibility-tests || touch FAIL
+    store_test_results "tests/compatibility-tests-results" "${compatibility_dir}"
+    [[ ! -f FAIL ]] || die "compatibility tests failed for Central v${short_central_tag}, Sensor v${short_sensor_tag}"
+
+    cd "$ROOT"
+
+    collect_and_check_stackrox_logs "/tmp/compatibility-test-logs" "${compatibility_dir}/initial_tests"
+}
+
+test_preamble() {
+    require_executable "roxctl"
+
+    export ROX_PLAINTEXT_ENDPOINTS="8080,grpc@8081"
+    export ROXDEPLOY_CONFIG_FILE_MAP="$ROOT/scripts/ci/endpoints/endpoints.yaml"
+    export TRUSTED_CA_FILE="$ROOT/tests/bad-ca/root.crt"
+}
+
```

Comment:
Potential followup: Remove code duplication, the same function exists in run.sh for the Go e2e tests

// Or just add a TODO comment for now